fe/fe-core/src/main/java/org/apache/doris/analysis/CreateTableLikeStmt.java

```diff
@@ -0,0 +1,130 @@
+// under the License.
+
+package org.apache.doris.analysis;
+
+import com.google.common.collect.Lists;
+import org.apache.doris.catalog.Catalog;
+import org.apache.doris.catalog.Database;
+import org.apache.doris.catalog.Table;
+import org.apache.doris.common.ErrorCode;
+import org.apache.doris.common.ErrorReport;
+import org.apache.doris.common.FeNameFormat;
+import org.apache.doris.common.UserException;
+import org.apache.doris.common.util.SqlParserUtils;
+import org.apache.doris.mysql.privilege.PrivPredicate;
+import org.apache.doris.qe.ConnectContext;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.util.List;
+
+/**
+ * @author wangcong
+ * @version 1.0
+ * @date 2020/10/7 10:32 上午
+ */
+public class CreateTableLikeStmt extends DdlStmt {
+    private static final Logger LOG = LogManager.getLogger(CreateTableLikeStmt.class);
+
+    private CreateTableStmt parsedCreateTableStmt;
+    private final boolean isExternal;
+    private final boolean ifNotExists;
+    private final TableName tableName;
+    private final TableName existedTableName;
+
+    public CreateTableLikeStmt(boolean ifNotExists, boolean isExternal, TableName tableName, TableName existedTableName) {
+        this.ifNotExists = ifNotExists;
+        this.isExternal = isExternal;
+        this.tableName = tableName;
+        this.existedTableName = existedTableName;
+    }
+
+    public boolean isSetIfNotExists() {
+        return ifNotExists;
+    }
+
+    public boolean isExternal() {
+        return isExternal;
+    }
+
+    public String getDbName() {
+        return tableName.getDb();
+    }
+
+    public String getTableName() {
+        return tableName.getTbl();
+    }
+
+    public String getExistedDbName() {
+        return existedTableName.getDb();
+    }
+
+    public String getExistedTableName() {
+        return existedTableName.getTbl();
+    }
+
+    public CreateTableStmt getParsedCreateTableStmt() {
+        return parsedCreateTableStmt;
+    }
+
+    @Override
+    public void analyze(Analyzer analyzer) throws UserException {
+        super.analyze(analyzer);
+        existedTableName.analyze(analyzer);
+        ConnectContext ctx = ConnectContext.get();
+        if (!Catalog.getCurrentCatalog().getAuth().checkTblPriv(ctx, existedTableName.getDb(),
+                existedTableName.getTbl(), PrivPredicate.SELECT)) {
+            ErrorReport.reportAnalysisException(ErrorCode.ERR_SPECIFIC_ACCESS_DENIED_ERROR, "SELECT");
+        }
+
+        tableName.analyze(analyzer);
+        FeNameFormat.checkTableName(getTableName());
+        if (!Catalog.getCurrentCatalog().getAuth().checkTblPriv(ctx, tableName.getDb(),
+                tableName.getTbl(), PrivPredicate.CREATE)) {
+            ErrorReport.reportAnalysisException(ErrorCode.ERR_SPECIFIC_ACCESS_DENIED_ERROR, "CREATE");
+        }
+
+        Database db = Catalog.getCurrentCatalog().getDb(getExistedDbName());
```

Comment:
line 102 to line 118 is better moved to `createTableLike()` in Catalog.java.
Because analysis phase is usually used to analyze grammar, not semantics.

Reply:
done